internal/factsengine/gatherers/saphostctrl_test.go

```diff
@@ -0,0 +1,129 @@
+package gatherers // nolint:dupl
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	utilsMocks "github.com/trento-project/agent/pkg/utils/mocks"
+)
+
+type SapHostCtrlTestSuite struct {
+	suite.Suite
+	mockExecutor *utilsMocks.CommandExecutor
+}
+
+func TestSapHostCtrlTestSuite(t *testing.T) {
+	suite.Run(t, new(SapHostCtrlTestSuite))
+}
+
+func (suite *SapHostCtrlTestSuite) SetupTest() {
+	suite.mockExecutor = new(utilsMocks.CommandExecutor)
+}
+
+func (suite *SapHostCtrlTestSuite) TestSapHostCtrlGather() {
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "Ping").Return(
+		[]byte("SUCCESS (  543341 usec)\n"), nil)
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "ListInstances").Return(
+		[]byte(" Inst Info : PRD - 00 - myhost-vmhana02 - 753, patch 410, changelist 1908545\n"), nil)
+
```

Comment:
I would add other scenario when there is not any instance, to see that it returns an empty map

Reply:
And we're probably missing a test for a failed `ListInstances` as well